edd-discount-widget.php

```diff
@@ -55,8 +55,14 @@ function widget( $args, $instance ) {
             if( $title ) echo $before_title . $title . $after_title;
 
             $discounts = $this->get_discounts( $site_url, $api_key, $api_token, $max_discounts );
-            $discounts = json_decode( $discounts, true );
-            $discounts = $discounts['discounts'];
+
+            // A string is only returned in the event of an error.
+            if (is_string($discounts)) {
```

Comment:
Instead of a string for errors, how about either:

- A `WP_Error` object (then we can check if `is_wp_error()` and then get the error message; or
- Throw an exception on errors. Then we can catch that and display the error message.

Additionally, I'm thinking maybe only admins should actually see the error messages. Otherwise guests should just see nothing.

Reply:
I like the idea of exceptions but I think I'll do a WP_Error, makes sense.